BrowserKit/Sources/OnboardingKit/Views/Common/OnboardingBottomSheetViewController.swift

```diff
@@ -0,0 +1,185 @@
+        closeButton.scalesLargeContentImage = true
+
+        addChild(child)
+        view.addSubviews(backgroundView, contentView, closeButton)
+        view.accessibilityElements = [closeButton, contentView]
+        contentView.addSubview(child.view)
+        child.view.translatesAutoresizingMaskIntoConstraints = false
+
+        backgroundView.pinToSuperview()
+        NSLayoutConstraint.activate([
+            closeButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -UX.closeButtonPadding),
+            closeButton.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor,
+                                             constant: UX.closeButtonPadding),
+
+            contentView.topAnchor.constraint(equalTo: closeButton.bottomAnchor),
+            contentView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            contentView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            contentView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+
+            child.view.topAnchor.constraint(equalTo: contentView.contentLayoutGuide.topAnchor),
+            child.view.bottomAnchor.constraint(equalTo: contentView.contentLayoutGuide.bottomAnchor),
+            child.view.leadingAnchor.constraint(equalTo: contentView.contentLayoutGuide.leadingAnchor),
+            child.view.trailingAnchor.constraint(equalTo: contentView.contentLayoutGuide.trailingAnchor),
+            child.view.widthAnchor.constraint(equalTo: contentView.frameLayoutGuide.widthAnchor),
+
+            child.view.heightAnchor.constraint(greaterThanOrEqualTo: contentView.heightAnchor)
+        ])
+
+        child.didMove(toParent: self)
+        closeButton.setContentCompressionResistancePriority(.required, for: .vertical)
+        closeButton.setContentHuggingPriority(.required, for: .vertical)
+    }
+
+    private func calculateAndUpdateDetentsHeight() {
+        let targetSize = CGSize(
+            width: view.bounds.width,
+            height: UIView.layoutFittingCompressedSize.height
+        )
+
+        let fittingSize = child.view.systemLayoutSizeFitting(
+            targetSize,
+            withHorizontalFittingPriority: .required,
+            verticalFittingPriority: .fittingSizeLevel
+        )
+        closeButton.layoutIfNeeded()
+        let calculatedHeight = fittingSize.height
+                                + closeButton.frame.height
+                                + UX.closeButtonPadding * 2.0
+        if #available(iOS 16.0, *) {
+            lastCalculatedHeight = calculatedHeight
+            sheetPresentationController?.animateChanges { [weak self] in
+                self?.sheetPresentationController?.invalidateDetents()
+            }
+        }
+    }
+
+    // MARK: - Notifiable
+    nonisolated public func handleNotifications(_ notification: Notification) {
+        DispatchQueue.main.async { [self] in
+            calculateAndUpdateDetentsHeight()
```

Comment:
Let's check for the correct notification here so we don't recalculate unnecessarly.